### PR TITLE
[DT-1127] Improve error message for column incorrect type

### PR DESCRIPTION
**Pull request by exyi**
We now explicitly handle the case when the column has an unexpected type.

We also do implicit conversions to Double and Long from other numeric types.

Minor improvements: shorter type names, hint for parentExternalId=NULL

### CODE REVIEW

**Comment by exyi** [comment]
I also had to make some  to the tests (I mean the second commit), since some data are broken again... All were also needed on master to pass.

File: macro/src/main/scala/SparkSchemaHelper.scala [hunk 1]
```diff
@@ -80,16 +80,54 @@ class SparkSchemaHelperImpl(val c: Context) {
...
 
-        val column = q"scala.util.Try(Option($r.getAs[$rowType]($name))).toOption.flatten"
+        val throwError =
+          q"throw cognite.spark.v1.SparkSchemaHelper.badRowError($r, $name, ${rowType.toString}, ${structType.toString})"
+
+        def handleFieldValue(value: Tree) =
+          if (rowType <:< typeOf[Double]) {
+            // do implicit conversion to double
+            q"""($value match {
+             case x:Double => Some(x)
```
Comment (wjoel) [inline macro/src/main/scala/SparkSchemaHelper.scala:90]: Not super important but there should be a space after the colon ;)

File: src/test/scala/cognite/spark/v1/AssetsRelationTest.scala [hunk 2]
```diff
@@ -391,14 +391,15 @@ class AssetsRelationTest extends FlatSpec with Matchers with SparkTest {
         .write
         .insertInto("destinationAssetsUpsert")
 
+      val assetsCreated = getNumberOfRowsCreated(metricsPrefix, "assets")
```
Comment (wjoel) [inline src/test/scala/cognite/spark/v1/AssetsRelationTest.scala:394]: Do you have any idea why/if moving this check here makes a difference?

Comment (exyi) [reply src/test/scala/cognite/spark/v1/AssetsRelationTest.scala:394]: It does not fix the test, just asserts the easy to  assert and easy to debug stuff first.

**Pull request merged**